kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/throwable/matchers.kt

```diff
@@ -0,0 +1,83 @@
+package io.kotlintest.matchers.throwable
+
+import io.kotlintest.Matcher
+import io.kotlintest.Result
+import io.kotlintest.should
+import io.kotlintest.shouldNot
+
+inline fun <reified T : Throwable> Throwable.shouldBeOfType() = this should beOfType<T>()
+inline fun <reified T : Throwable> Throwable.shouldNotBeOfType() = this shouldNot beOfType<T>()
+inline fun <reified T : Throwable> beOfType() = object : Matcher<Throwable> {
+  override fun test(value: Throwable) = Result(
+      value is T,
+      "Value should be of type ${T::class}, but instead got ${value::class}",
+      "Value should not be of type ${T::class}"
+  )
+}
+
+inline fun <reified T : Throwable> Throwable.shouldBeExactlyOfType() = this should beOfTypeExactly<T>()
```

Comment:
`obj.shouldBeTypeOf<T>` covers this already

Reply:
This is to check if the classes are exaclty the same. If you use only `shouldBeOfType` and throw a `FileNotFoundException`, it will accept `IOException`. This is to verify the exactly same class.

Reply:
I don't think it will.
Unless our docs are lying xP

| Types ||
| ------- | ---- |
| `obj.shouldBeSameInstanceAs(other)` | Compares objects by identity, that is, they are the same exact reference. |
| `obj.shouldBeTypeOf<T>()` | Asserts that the given reference is exactly of type T. Subclass will fail. Ie, `1 should beOfType<Number>` would fail because although 1 _is_ a Number, the runtime type is not Number. |
| `obj.shouldBeInstanceOf<T>` | Asserts that the given reference is of type T or a subclass of T. |
| `obj.shouldHaveAnnotation(annotationClass)` | Asserts that the object has an annotation of the given type. |
| `obj.shouldBeNull()` | Asserts that a given reference is null.|

Reply:
`ShouldBeTypeOf` shouldn't allow for subclasses, while `shouldBeInstanceOf` should

Reply:
Oh, right, changed both of them :man_facepalming:

Reply:
done!